### DISCUSSION TITLE
Unchecked checkbox isn't in requestData?

### DISCUSSION THREAD

**Discussion by karam94** [adonisjs/core#1532] [Help]
```
<label for="isRestaurant" class="checkbox">
   <input type="checkbox" name="isRestaurant">
    Yes
</label>
```

I have this checkbox amongst other similar form items.
According to the Adonis 5 docs, it should be submitting 'on' or 'off' as part of the request object to my controller.

When the checkbox is checked and I do `console.log(request[requestData])`, I see `isRestaurant: 'on'`.

However, when the checkbox is left unchecked, `isRestaurant` does not exist on my request.

This is causing me issues because I want this to be persisted as either true or false, not NULL within my database.

Does Adonis not handle the post of an unchecked checkbox? Meaning we get the default behaviour where an unchecked checkbox isn't sent on the post? If so, what is the recommended approach in the controller to check whether `isRestaurant` is on the request with a validator?

Make it optional on the validator and then do `!!validatedPost.isRestaurant` within my `await Post.create({...})`?

Thanks

**Comment by thetutlage** [comment]
Thats how HTML works. Here's a related Stack overflow thread https://stackoverflow.com/questions/1809494/post-unchecked-html-checkboxes

**Reply by karam94** [reply]
> Thats how HTML works. Here's a related Stack overflow thread https://stackoverflow.com/questions/1809494/post-unchecked-html-checkboxes

Yeah, I know that.
But I was trying to ascertain if there was a cleaner way to do this and if my proposed alternative solution was feasible.

Anyways, for future reference if any one else comes across this, I've just done it like this:
To avoid the two inputs, I've ensured the `isRestaurant` is optional on my validator:
`isRestaurant: schema.boolean.optional(),`

After the validator stage, when creating the object, I then map it as so:
`isRestaurant: !!validatedPost.isRestaurant,`
This way, if it is null, it is false, otherwise it is true and this solves the issue to avoid persisting NULL to the SQL database.
